scipy/stats/_axis_nan_policy.py

```diff
@@ -307,11 +376,19 @@ def hypotest_fun(x):
                     if np.isnan(x).any():
                         return result_object(np.nan, np.nan)
                     samples = np.split(x, split_indices)[:n_samp]
+                    if sentinel:
+                        samples = _remove_sentinel(samples, paired, sentinel)
+                    if is_too_small(samples):
+                        return result_object(np.nan, np.nan)
```

Comment:
We were already doing this `is_too_small` check for `nan_policy == 'omit'`; it's for when there are too few values (e.g. zero) left in the axis-slice to perform the function.